File: Data Structures & Algorithms/permutation-string/submission-0.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int[] count=new int[26];
        int left=0;
        for(int i=0;i<s1.length();i++){
            count[s1.charAt(i)-'a']++;
        }
        for(int right=0;right<s2.length();right++){
            count[s2.charAt(right)-'a']--;
            while(right-left+1>s1.length()){
                count[s2.charAt(left)-'a']++;
                left++;
            }
            if(right-left+1==s1.length()){
                if(mismatch(count)){
                    return true;
                }
            }
        }
        return false;
    }
    private static boolean mismatch(int[] count){
        for(int i=0;i<26;i++){
            if(count[i]!=0){
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        queue=deque([root])
        lis=[]
        while queue:
            curr_list=[]
            curr_len=len(queue)
            for _ in range(curr_len):
                curr=queue.popleft()
                curr_list.append(curr.val)
                if curr.left:
                    queue.append(curr.left)
                if curr.right:
                    queue.append(curr.right)
            lis.append(curr_list)
        return lis


